internal/configuration/provider.go

```diff
@@ -0,0 +1,175 @@
+package configuration
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/knadh/koanf"
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/authelia/authelia/internal/configuration/schema"
+	"github.com/authelia/authelia/internal/configuration/validator"
+)
+
+// Provider holds the koanf.Koanf instance and the schema.Configuration.
+type Provider struct {
```

Comment:
I think introducing a provider could be a good idea here even though I would call it a ConfigurationEnricher or something like that but I don't really get the objective of this one. I mean there are many public functions doing side effects, I feel it's hard to know how to use it.
I would expect this struct to take the unstructured pieces of configuration and the validator as input and produce a structured configuration or throw an error. So I would expect one or two functions from the API to be public at most but the rest should be private imo.